logical_router.go

```diff
@@ -330,20 +329,16 @@ func (odbi *ovndb) lrlbListImp(lr string) ([]*LoadBalancer, error) {
 						if err != nil {
 							return nil, err
 						}
-						listLB = append(listLB, lb)
+						return []*LoadBalancer{lb}, nil
 					} else {
 						return nil, fmt.Errorf("type libovsdb.UUID casting failed")
 					}
 				default:
 					return nil, fmt.Errorf("Unsupport type found in ovsdb rows")
 				}
 			}
-			lrFound = true
-			break
+			return []*LoadBalancer{}, nil
```

Comment:
this breaks older behaviour - before changes if list return empty slice it returns ErrorNotFound

Reply:
@vtolstov Can you please clarify? In the old code, if lrFound is true, it returns empty list. It only returns ErrorNotFound if the specific logical router is not found, which is the case in the new code as well.

Reply:
I revisited this and it seems @vtolstov  is right. We can't just return because for each of the entities in for loop, there are two cases, ovsset and uuid. Hence, static slice capacity will not help here. We need to use dynamic slice  which is len(go.set)+ uuids for which we don't know how many are in place. We return entire slice accordingly . Hence, (keep the current code as is). What do you think? Please feel free to extend a UT for the same.

Reply:
@noah8713 Here is my understanding: that for loop is only used to find the right router, and if it finds the right router, it breaks. We don't continue the for loop to find more entries. So once the right router is found, there are only two cases: ovsset or uuid. In both cases, we can preallocate the right size of slice.

Please let me know if my understanding is incorrect. Thanks!

Reply:
Ah I see, I guess I got little confused again reading size as uuid + go.set as I saw @vtolstov comment to see if I didn't miss anything. LGTM again :)

Reply:
thanks!